src/view/user/setting/StudyList/query.ts

```diff
@@ -0,0 +1,68 @@
+import { getGroupList, patchGroupVisibility } from "@/api/groups";
+import type { GroupListItem, GroupStatus } from "@/api/groups/type";
+import { useToast } from "@/common/hook/useToast";
+import { HTTP_ERROR_STATUS } from "@/shared/constant/api";
+import type { StudyListType } from "@/view/user/setting/StudyList/StudyListTable/type";
+import {
+  useMutation,
+  useQueryClient,
+  useSuspenseQuery,
+} from "@tanstack/react-query";
+import type { HTTPError } from "ky";
+
+export const useGetMyGroupsQuery = () => {
+  return useSuspenseQuery({
+    queryKey: ["groups", "setting"],
+    queryFn: getGroupList,
+    select: (data) =>
+      [
+        ...transformData(data.bookmarked, "bookmarked"),
+        ...transformData(data.queued, "queued"),
+        ...transformData(data.inProgress, "inProgress"),
+        ...transformData(data.done, "done"),
+      ].sort((a, b) => a.id - b.id),
```

Comment:
```suggestion
    ["bookmarked", "queued", "inProgress", "done"]
        .flatMap((status) => transformData(data[status], status))
        .sort((a, b) => a.id - b.id),
```

이런 식으로 리팩토링 해볼 수 있을 것 같네요!

Reply:
오호 `flatMap`을 사용할 수도 있네요 !! 좋습니다